src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +427,69 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
```

Comment:
> PRIMARY KEY (site_id, user_id, role_id)

This is not correct. If using composite primary key, it must be `PRIMARY KEY (site_id, user_id)`, as users shouldn't have two or more roles for the one site:
```
| Site        | User  | Role  |
| ----------- | ----- | ----  |
| example.com | John  | Admin |
| example.com | Jane  | User  |
| example.com | Jane  | Admin | Incorrect. Duplicated primary key.
| example.org | Jane  | Admin |
```